packages/console/src/components/ConfirmModal/ConfirmPromiseModal.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { InstanceProps } from 'react-modal-promise';
+
+import ConfirmModal, { ConfirmModalProps } from './ConfirmModal';
+
+type ConfirmPromiseModalProps = Omit<
+  ConfirmModalProps & InstanceProps<boolean>,
+  'onCancel' | 'onConfirm'
+>;
+
+const ConfirmPromiseModal = ({ onResolve, onReject, ...rest }: ConfirmPromiseModalProps) => (
+  <ConfirmModal
```

Comment:
I think we don't have to maintain two confirm modal machanisms. Suggesting merging these `<ConfirmPromiseModal />` and `<ConfirmModal />` components into one.